baseline-network-linux-hld.md

```diff
@@ -0,0 +1,163 @@
+networking hardware support, without addressing the question of how to abstract
+and control the switch forwarding plane itself.
+
+The intention of this project is not to replace or compete with any of the
+existing network operating systems, but rather to provide a common and
+extensible layer on which those can be built. Its principles and paradigms are
+largely aligned and adapted from the
+[OCP OpenNetworkLinux Project](https://github.com/opencomputeproject/OpenNetworkLinux).
+This proposal specifically seeks an agreement between the developers and
+maintainers of the aforementioned network operating system communities, to be
+able to shared common implementations to support open networking hardware.
+
+Support for open networking hardware platforms is currently distributed over
+various different projects. Many platforms, especially older ones are supported
+by OpenNetworkLinux, many are supported by SONiC, and some are supported by
+DentOS, which is based on OpenNetworkLinux, but targets different devices. The
+open hardware support is usually very tightly integrated in the build systems,
+which makes sharing and reusing the support hard.
+
+## Requirements and Dependencies
+
+## Functional Description
+
+### Base Open Networking Hardware Support
+
+The base networking hardware support combines hardware support from multiple
+sources, and therefore will need to support multiple kinds of APIs.
+
+Possible APIs are:
+
+- SONiC PDDF, due to the large amount of available platforms
+  https://github.com/sonic-net/SONiC/blob/master/doc/platform/brcm_pdk_pddf.md
+- S3IP as described in https://github.com/sonic-net/SONiC/pull/1068
+- ONLP, as used by ONL and DentOS
+
+The base networking hardware support is organized in multiple repositories:
+
+- kernel: Linux fork with required base changes (backports of important
+submissions, etc)
+- platform-support: drivers, libraries and userspace code for initializing
+platforms
+(do we need more?)
+- TBD: organization of platform support
+- TBD: platform initialization
+
+In the first iteration, the platform support is fine as-is. Future iterations
+should focus on improving the platform drivers and support code to follow modern
+code style and compile with as few warnings as possible, ideally none.
+
+New platform support should follow modern coding styles from the beginning (e.g.
+kernel coding style for modules).
+
+## Design
+
+![example layering with operating layers on top like meta-dent, the proposed
+hardware support layer in the middle, and base Yocto and Open Embedded layers
+at the bottom](images/layering-example.png "layering example")
+
+### Yocto based example and testing OS
+
+To allow quick testing and easy extension for other users, an example
+implementation and support layer for Yocto will be provided. It will contain all
+basic building blocks for creation of an installable image:
+
+- recipes for above kernel and platform support
+- class and image recipe for a ONIE installable image
```

Comment:
```suggestion
- class and image recipe for an ONIE installable image
```

Reply:
ONIE is mentioned here first. It should be spelled out and at least a link should be provided.